app/assets/javascripts/host_checkbox.js.erb

```diff
@@ -115,12 +146,26 @@ $(function() {
 function submit_modal_form() {
   if (!$('#keep_selected').is(':checked'))
     removeForemanHostsCookie();
+  if ($("#multiple-chosen").data('multiple')){
+    var query = $("<input>")
+            .attr("type", "hidden")
+            .attr("name", "search").val($("#search").val());
+    $("#confirmation-modal form").append($(query));
+  }
+
   $("#confirmation-modal form").submit();
   $('#confirmation-modal').modal('hide');
 }
 
 function build_modal(element, url) {
-  var url = url + "?" + $.param({host_ids: $.foremanSelectedHosts});
+  if ($("#multiple-chosen").data('multiple')) {
+    var query = $("<input>")
```

Comment:
why do you need to create an input here if the value is encoded directly to the query?